src/ui/classic/waylandpointer.cpp

```diff
@@ -62,4 +69,28 @@ void WaylandPointer::initPointer() {
     });
 }
 
+void WaylandPointer::initTouch() {
+    touch_->down().connect(
+        [this](uint32_t, uint32_t, wayland::WlSurface *surface, int, wl_fixed_t sx, wl_fixed_t sy) {
+            // TODO handle id for multiple touch
+            auto *window = static_cast<WaylandWindow *>(surface->userData());
+            if (!window) {
+                return;
+            }
+            focus_ = window->watch();
```

Comment:
touch focus should not be the same as pointer foucs. A new focus member that is only used by touch events should be defined.

Reply:
Thank you!
I have made new focus members for touch events.